### DISCUSSION TITLE
能不能把网关塞到op主路由中？

### DISCUSSION THREAD

**Discussion by yyysuo** [kkkgo/PaoPaoGateWay#55] [问和答]
感谢大佬提供了这么好的分流思路，目前没用paopao的方案，自己在esxi下用主路由op+clash vm/sb vm，实现了fakeip网关，体验非常棒。
但是目前的方案得用2个vm，能否在主路由op中跑mihomo core或者sb core，tun模式关闭auto-router，静态路由转发到fakeip到tun ip段呢？比如fakeip段为198.18.0.1/16，sing-box入站tun配置为
    }, 
    "inbounds": [ 
        { 
            "type": "tun", 
            "tag": "tun-in", 
            "inet4_address": "172.18.0.1/30", 
            "mtu": 9000, 
            "auto_route": true, 
            "strict_route": true, 
            "stack": "system", 
            "sniff": true, 
            "sniff_override_destination": false 
        } 

试了一下，sing-box跑在主路由op上，dns的入站是正常的，客户端也能正常的收到fakeip，但是添加静态路由转发fakeip段到tun ip 172.18.0.1，sing-box面板里面看不到入站的连接。

问题：这种思路是否是可行的？

**Comment by kkkgo** [comment]
如果你不把主路由和fakeip网关的拓扑分开的话，那么就是传统的all in one方案了，跟openclash之类没啥区别。

**Reply by yyysuo** [reply]
还是有区别的吧，根据大佬的文章，传统的方案是所有流量劫持，再分流，用的是iptables nftables等手段。这样搞并不是劫持所有流量，并且也是静态路由，本质还是域名分流，都在本机，理论上快一丝？

**Reply by kkkgo** [reply]
如果是在“本机”搞，那么用dns+ipset方案更好。把解析出来的结果丢进ipset集合，然后改集合走代理。

**Reply by yyysuo** [reply]
> 如果是在“本机”搞，那么用dns+ipset方案更好。把解析出来的结果丢进ipset集合，然后改集合走代理。

ipset比静态路由效率还高么，用ipset，真和passwall和ssrp没区别了，因为他们也是用ipset

**Reply by kkkgo** [reply]
静态路由的应用场景本来就是非本机。本机虚拟接口效率比不上策略路由。

**Reply by yyysuo** [reply]
感谢大佬解惑。
